iep-008/README.adoc

```diff
@@ -0,0 +1,155 @@
+The objective here, is to dockerize this application in order to run it on a kubernetes cluster.
+
+As a stateful application, we must take into consideration following aspect when deploying the ldap server in production.
+
+=== Access
+Ldap must be accessible from inside and outside the kubernetes cluster.
+In order to reach ldap from the wild, we need a fixed public IP and only allow connection from whitelisted IP.
+This can be easily done with the kubernetes resource 'service'.
+
+[cols="1a,2a", options="header"]
+.Access
+|===
+|Inside
+|Outside
+|
+* https://accounts.jenkins.io/[Accountapp]
+|
+* https://repo.jenkins-ci.org/webapp/#/home[Artifactory]
+* https://ci.jenkins.io[ci.jenkins.io]
+* https://wiki.jenkins.io/[Confluence]
+* https://issues.jenkins-ci.org[Jira]
+* puppet master
+* spambot
+* trusted.ci
+|===
+
+
+=== Backup/Restore
+The procedure to backup/restore the database should be easy, two scripts are provided by the same docker image that run ldap. +
+Backups must be stored on an Azure File Storage in order to simplify their access from various location. +
+Backup name must respect format 'YYYYmmddHHMM'
+
+Backups and restore operation must be done in following situations:
+
+[cols="1a,2a", options="header"]
+.Backup/Restore
+|===
+|Backup
+|Restore
+| * On daily basis
+* When the application is stopping
+* On demand
+| * On demand
+|===
+
+=== Certificate
+I didn't find an easy way to use Letsencrypt certificate from Kube-lego, the constraint that I have at the moment is
+ingress resources and service resources don't/can't use the same public IP.
+
+=== Data
+The ldap database must be store on a stable storage that can be easily mounted/unmounted.
+Currently there are no perfect solutions as each solution has advantages and disadvanteges.
+
+==== Dedicated Azure Disk storage
+ReadWriteOnce
+[cols="1a,2a", options="header"]
+.Dedicated Azure Disk Storage
+|===
+|+
+|-
+|
+* Persistent Data across kubernetes clusters as we only have one container running at the time.
+* We only have to restore a backup once.
+|
+* Complexify cluster upgrade, https://github.com/jenkins-infra/iep/tree/master/iep-007[iep-007],
+traffic will be redirected to the new server once the old is delete.
+* It means downtime when we upgrade the container as we must delete the old container before starting the new one.
+|===
+
+==== Dynamic Azure Disk Storage
+ReadWriteOnce
+
+[cols="1a,2a", options="header"]
+.Dynamic Azure Disk Storage
+|===
+|+
+|-
+|
+* Persistent data associated to a cluster lifecycle
+* Simplify cluster migration, new cluster can be started even if the old cluster is still running
+|
+* We must restore a backup on each new kubernetes cluster deployment
+* While migrating the cluster, we must be sure to put the old cluster in read only mode.
+|
+|===
+
+==== Azure File storage
+ReadWriteMany +
+Unfortunately OpenLdap doesn't run well on CIFS partition, so this solution doesn't work at the moment.
```

Comment:
I can  think of a lot of reasons why OpenLDAP might not run well on CIFS, could you add an explanation of _why_, and perhaps a footnote to wherever you may have read that?

That will be helpful for new folks in the future